chia/timelord/timelord_launcher.py

```diff
@@ -74,14 +84,15 @@ async def spawn_process(host: str, port: int, counter: int, lock: asyncio.Lock,
                     first_10_seconds = False
             else:
                 log.error(f"VDF client {counter}: {stderr.decode().rstrip()}")
-        log.info(f"Process number {counter} ended.")
-        async with lock:
-            if proc in active_processes:
-                active_processes.remove(proc)
+
+        async with process_mgr.lock:
+            if proc in process_mgr.active_processes:
+                process_mgr.active_processes.remove(proc)
+
```

Comment:
It feels like this and other stuff should be wrapped up by the class.  Basically, instead of providing a locking mechanism and hoping folks use it, apply the locking mechanism yourself with helpful abstractions that can be called instead.

Or maybe this whole function is a method maybe?